src/actions/WalletActions.tsx

```diff
@@ -312,3 +316,134 @@ const activateWalletTokens = async (
     throw new Error('Activation with multiple wallet options not supported yet')
   }
 }
+
+const asInfoServerResponse = asObject({
+  docs: asArray(asUnknown)
+})
+
+const asKeyInfo = asObject({
+  pubKeyHash: asString,
+  exposed: asBoolean
+})
+
+type KeyInfo = ReturnType<typeof asKeyInfo>
+
+export function checkCompromisedKeys(navigation: NavigationBase): ThunkAction<Promise<void>> {
+  return async (dispatch, getState) => {
+    const state = getState()
+    const account = state.core.account
+
+    const { activeWalletIds, currencyWallets } = account
+
+    // Get synced setting Settings.json with public key map securityCheckedWallets: {walletId: { checked: boolean, modalShown: number }}
+    const settings = await getSyncedSettings(account)
+    let securityCheckedWallets = { ...settings.securityCheckedWallets }
+
+    // Gather list to send to info server
+    const hashedPubKeys = new Map<string, string>()
+    for (const walletId of activeWalletIds) {
+      // Add entries for walletIds that aren't recognized
+      if (securityCheckedWallets[walletId] == null) {
+        securityCheckedWallets = { ...securityCheckedWallets, [walletId]: { checked: false, modalShown: 0 } }
```

Comment:
Since you have already made a copy, I would just mutate it, like `securityCheckedWallets[walletId] = { checked: false }`. Otherwise it's O(n^2) to keep re-copying the object every time around the loop.